### DISCUSSION TITLE
Dynamic Links not being evaluated correctly after server-side redirect

### DISCUSSION THREAD

**Discussion by ca057** [invertase/react-native-firebase#4419] [Q&A] [answered]
Hello,

I hope someone can help out or has made a similar experience which might help me to understand what goes wrong. I’m using dynamic links in an app to support different workflows for the user to manage their own account. In one of them I send them an email, where they have to click a link to my own backend where I perform several checks based on the link (imagine a verification link). If everything goes well, I issue a redirect (HTTP 307) to a dynamic link which should push my users back into the app.
Unfortunately this is not going as expected. On Android it’s only working smoothly if I’m logged into my google account in my default browser, otherwise I end up in the Google PlayStore.
On iOS it’s only working if I enable the "app preview page" or (if I’m on iOS 14 have set Chrome as my default browser - I guess any other browser than Safari would be sufficient). I would like to not use the "app preview page" and bring the users quickly back into my app.
Using dynamic links directly works as expected (which makes sense).

Has anyone worked with a similar setup or can guide me into a direction how I can make this work or why it can’t work at all anyway? Is there something I have to change about my redirect?

Is there more information you need to understand the issue better?

I’m still running the following versions:
* react-native: 0.62.2
* @react-native-firebase/app: 6.3.4
* @react-native-firebase/dynamic-links: 6.3.4

Thanks for the help,
Christian

**Comment by mikehardy** [comment, marked as answer]
With apologies, this will be an upstream issue and have nothing to do with react-native-firebase, it will be firebase-ios-sdk or firebase-android-sdk and you'll need to do a quickstart there to reproduce it and ask for upstream attention. From what I understand (and have used myself) dynamic links are an interesting feature when they work, but relying on them for authentication etc is not something I would ever do. It relies on (for example) correct download and installation in a timely manner of applinks files from iOS, correct browser behavior etc etc.

**Comment by ca057** [comment]
Thanks @mikehardy for your quick answer - I assumed this would be the case. I had hoped someone here used it in a similar way and could share their experience 😊
